ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/SharedApiTypes.java

```diff
@@ -55,6 +62,52 @@ public class SharedApiTypes {
                   GetGenesisApiDataBuilder::genesisForkVersion)
               .build());
 
+  private static final DeserializableTypeDefinition<ProposerDuty> PROPOSER_DUTY_TYPE =
+      DeserializableTypeDefinition.object(
+              ProposerDuty.class, ProposerDuty.ProposerDutyBuilder.class)
+          .initializer(ProposerDuty.ProposerDutyBuilder::new)
+          .finisher(ProposerDuty.ProposerDutyBuilder::build)
+          .withField(
+              "pubkey",
+              PUBLIC_KEY_TYPE,
+              ProposerDuty::getPublicKey,
+              ProposerDuty.ProposerDutyBuilder::publicKey)
+          .withField(
+              "validator_index",
+              INTEGER_TYPE,
+              ProposerDuty::getValidatorIndex,
+              ProposerDuty.ProposerDutyBuilder::validatorIndex)
+          .withField(
+              "slot", UINT64_TYPE, ProposerDuty::getSlot, ProposerDuty.ProposerDutyBuilder::slot)
+          .build();
```

Comment:
I wonder if this will be less messy if this lives in the type, i think the SharedApiTypes class will scale very badly if we throw things into it for the entire move